Update v1.2 release notes (#58)
osisoft/PI-Adapter-OPC-UA-Docs: content/release-notes/release-notes.md

# Release notes

PI Adapter for OPC UA 1.2

## Overview

> PI Adapter for OPC UA collects time series data and relevant metadata from an OPC UA (OPC Unified Architecture) server and sends it to configured OMF endpoints such as PI Web API and OSIsoft Cloud Services. PI Adapter for OPC UA can also collect health and diagnostics information. It supports buffering, unsolicited data collection, on-demand discovery of available data items on a data source, on-demand/automatic history recovery of data items, and various Windows and Linux-based operating systems as well as containerization.

Reviewer: ```suggestion
PI Adapter for OPC UA collects time series data and relevant metadata from an OPC UA (OPC Unified Architecture) server and sends it to configured OMF endpoints such as PI Web API and OSIsoft Cloud Services. PI Adapter for OPC UA can also collect health and diagnostics information. It supports buffering, unsolicited data collection, on-demand discovery of available data items on a data source, on-demand or automatic history recovery of data items, and various Windows and Linux-based operating systems as well as containerization.
```

> This represents the release version 1.2.0 for the PI Adapter for OPC UA. This product collects time series data and relevant metadata from an OPC UA (OPC Unified Architecture) server and sends it to configured OMF endpoints such as PI Web API and OSIsoft Cloud Services. PI Adapter for OPC UA can also collect health and diagnostics information. It supports buffering, unsolicited data collection, automatic discovery of available data items on a data source, on-demand/automatic history recovery of data items supporting HistoryReady user access level, and various Windows and Linux-based operating systems as well as containerization.

Reviewer 2: I believe that we could consider removing the first sentence and change the second to: PI Adapter for OPC UA collects time series...

and start with it.

Reviewer 3: Fixed. Thanks

> This represents the release version 1.2.0 for the PI Adapter for OPC UA. This product collects time series data and relevant metadata from an OPC UA (OPC Unified Architecture) server and sends it to configured OMF endpoints such as PI Web API and OSIsoft Cloud Services. PI Adapter for OPC UA can also collect health and diagnostics information. It supports buffering, unsolicited data collection, automatic discovery of available data items on a data source, on-demand/automatic history recovery of data items supporting HistoryReady user access level, and various Windows and Linux-based operating systems as well as containerization.

Reviewer 2: We might change the automatic discovery to on-demand discovery of available data items

Reviewer 3: Fixed. Thanks

> This represents the release version 1.2.0 for the PI Adapter for OPC UA. This product collects time series data and relevant metadata from an OPC UA (OPC Unified Architecture) server and sends it to configured OMF endpoints such as PI Web API and OSIsoft Cloud Services. PI Adapter for OPC UA can also collect health and diagnostics information. It supports buffering, unsolicited data collection, automatic discovery of available data items on a data source, on-demand/automatic history recovery of data items supporting HistoryReady user access level, and various Windows and Linux-based operating systems as well as containerization.

Reviewer 2: on-demand/automatic history recovery, and various Windows...

I wouldn't include this "of data items supporting HistoryReady user access level" that's for in dept description of feature

Reviewer 3: Fixed. Thanks

For more information see [PI Adapter for OPC UA overview](xref:PIAdapterForOPCUAOverview).

## Fixes and enhancements

### Fixes
The following issues reported from version 1.1 are fixed in this release.

> - Fixed the issue that timestamp and value migth not be correctly updated for the fast sampling rates.
> - Fixed the issue that browse operation can fail when empty result is returned from a server.

Reviewer: ```suggestion
- Timestamp and value might not be correctly updated for the fast sampling rates.
- Browse operation can fail when empty result is returned from a server.
```

### Enhancements
The following enhancements are added in this release.

> - Enhanced user experience to perform data source discovery.

Reviewer 2: On demand data source discovery feature has been added. or something along those lines

Reviewer 3: Fixed. Thanks

> - Added support for On-demand and automatic history recovery for data items that support HistoryReady user access level.

Reviewer 2: Here I would also stop with Added support for on on-demand and automatic history recovery.

Reviewer 3: Fixed. Thanks

> - Added on-demand data source discovery.
> - Added support for On-demand and automatic history recovery.
> - Added support for collection of Unit of Measure, Minimum and Maximum.
> - Added support for enumeration data types.
> - Added support for collecting data quality information along with data.

Reviewer: ```suggestion
- On-demand data source discovery.
- Support for On-demand and automatic history recovery.
- Support for collection of Unit of Measure, Minimum, and Maximum.
- Support for enumeration data types.
- Support for collecting data quality information along with data.
```

> - Added support for container creation with Unit of Measure, Min, Max, and Data Source properties.

Reviewer 2: Added support for collection of Unit of Measure, Minimum and Maximum.

Data source property was supported in the past.

Reviewer 3: Fixed. Thanks

> - Enabled data collection for enumeration data types from data items that have underlying enumeration types.

Reviewer 2: Added support for enumeration data types.

Reviewer 3: Fixed. Thanks

> - Added support for collecting data quality information along with the container data.

Reviewer 2: Adapter now collects data quality information.

Reviewer 3: Fixed. Thanks
